app/src/main/java/org/fossasia/openevent/app/core/auth/forgot/request/ForgotPasswordFragment.java

```diff
@@ -42,6 +42,11 @@ public View onCreateView(LayoutInflater inflater, ViewGroup container,
                              Bundle savedInstanceState) {
         binding = DataBindingUtil.inflate(inflater, R.layout.forgot_password_fragment, container, false);
         validator = new Validator(binding);
+
+        getPresenter().getEmailId().setEmail(
+            ((LoginFragment.EmailIdTransfer) getContext()).getEmail()
```

Comment:
Still unsafe, no casts should be there

Reply:
@iamareebjamal I checked [here](https://developer.android.com/training/basics/fragments/communicating) and found that SharedViewModel should be used. But they also had cast to their interface when using it. Why casting to the interface is unsafe?
What was the correct way?

Reply:
Explicitly setting interface reference from activity on fragment through `newInstance`

This solution is better